Xamarin.Forms.Core/Element.cs

```diff
@@ -208,9 +208,37 @@ public Element Parent
 				OnParentSet();
 
 				OnPropertyChanged();
+
+				if (this.IsTemplateRoot)
+					this.TemplatedParent = value;
 			}
 		}
 
+		internal event EventHandler TemplatedParentChanged;
+		
+		BindableObject _templatedParent;
+		public BindableObject TemplatedParent
+		{
+			get => _templatedParent ?? (_templatedParent = this.Parent?.TemplatedParent);
+			private set
+			{
+				if (_templatedParent == value)
+					return;
+				_templatedParent = value;
+				TemplatedParentChanged?.Invoke(this, null);
+				foreach (var element in this.Descendants())
+				{
+					if (value == null)
+						// Item has been detached from its templated 
+						// parent, so reset all the stored values
+						element._templatedParent = null;
+					element.TemplatedParentChanged?.Invoke(element, null);
```

Comment:
```suggestion
					element.OnPropertyChanged();
```
need to doublecheck this one ^^